app/main.py

```
@@ -349,73 +341,105 @@ def update_jdk_metrics():
+    
+    Parameters:
+        dt_str (str): Timestamp in the format "%Y%m%dT%H%M%S%z" (for example "20230101T123000+0000").
+    
+    Returns:
+        int: Epoch seconds for the parsed timestamp, or 0 if parsing fails or the input is invalid.
+    """
+    try:
+        dt = datetime.strptime(dt_str, "%Y%m%dT%H%M%S%z")
+        return int(dt.timestamp())
+    except (ValueError, TypeError):
+        logging.warning(f"Failed to parse timestamp: {dt_str!r}")
```

Ruff (0.15.4)

[warning] `warning()` call on root logger

(LOG015)